### PR TITLE
Fill in the codecs for day 2.

### PR DESCRIPTION
**Pull request by wjlow**
This covers: https://github.com/realestate-com-au/applied-scala/issues/12

### CODE REVIEW

File: src/main/scala/com/reagroup/appliedscala/models/MovieId.scala [hunk 1]
```diff
@@ -8,14 +8,17 @@ case class MovieId(value: Long)
...
     * {
     *   "id": 1
     * }
     *
-    * Hint: You don't want to use `deriveEncoder` here
+    * We don't want to use `deriveEncoder` here, otherwise the resulting JSON key will be tied to the name
+    * of the field in `MovieId`
     */
+
+  implicit val encoder: Encoder[MovieId] = Encoder.forProduct1("id")(_.value)
```
Comment (millyrowboat) [inline src/main/scala/com/reagroup/appliedscala/models/MovieId.scala:23]: I think we're not teaching `forProduct1` as much anymore in the Circe exercises, so might make more sense to use `Json.obj`

https://github.com/realestate-com-au/applied-scala/blob/solutions-20200728/src/main/scala/com/reagroup/appliedscala/models/MovieId.scala#L22

Comment (wjlow) [reply src/main/scala/com/reagroup/appliedscala/models/MovieId.scala:23]: I'll take that compromise 😄

Comment (wjlow) [reply src/main/scala/com/reagroup/appliedscala/models/MovieId.scala]: Done

**Review by nicholasnbg — APPROVED** [review]
Nice pickup including `MovieId`, I forgot it in the issue

**Pull request merged**